pkg/image/registry/imagesignature/rest.go

```diff
@@ -0,0 +1,98 @@
+package imagesignature
+
+import (
+	"errors"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	kapierrors "k8s.io/kubernetes/pkg/api/errors"
+	"k8s.io/kubernetes/pkg/api/rest"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/runtime"
+
+	imageapi "github.com/openshift/origin/pkg/image/api"
+	"github.com/openshift/origin/pkg/image/api/validation"
+	"github.com/openshift/origin/pkg/image/registry/image"
+)
+
+// REST implements the RESTStorage interface for ImageSignature
+type REST struct {
+	imageRegistry image.Registry
+}
+
+// NewREST returns a new REST.
+func NewREST(imageRegistry image.Registry) *REST {
+	return &REST{imageRegistry: imageRegistry}
+}
+
+// New is only implemented to make REST implement RESTStorage
+func (r *REST) New() runtime.Object {
+	return &imageapi.ImageSignature{}
+}
+
+func (r *REST) Create(ctx kapi.Context, obj runtime.Object) (runtime.Object, error) {
+	signature := obj.(*imageapi.ImageSignature)
+
+	if err := rest.BeforeCreate(Strategy, ctx, obj); err != nil {
+		return nil, err
+	}
+
+	imageName, _, err := imageapi.SplitImageSignatureName(signature.Name)
+	if err != nil {
+		return nil, kapierrors.NewBadRequest("ImageSignature name must be composed of <imageName>#<signatureName>")
```

Comment:
> This will be cleaner if handled by validation.

It's already handled by validation called from `BeforeCreate` few lines above. So this case shouldn't actually happen.